core/iwasm/common/wasm_shared_memory.c

```diff
@@ -87,6 +93,36 @@ search_module(WASMModuleCommon *module)
     return NULL;
 }
 
+static void
+create_list_of_waiter_addresses(void *key, void *value, void *user_data)
+{
+    AtomicWaitAddressArgs *data = (AtomicWaitAddressArgs *)user_data;
+    data->addr[(data->index++) % WAIT_MAP_SIZE] = key;
+}
+
+void
+notify_stale_threads_on_exception(WASMModuleInstanceCommon *module_inst)
+{
+    AtomicWaitAddressArgs *args = NULL;
+    if (!(args = wasm_runtime_malloc(sizeof(AtomicWaitAddressArgs)))) {
+        return;
```

Comment:
Should we print an error here for debugging purposes?

Reply:
added